numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -468,6 +471,18 @@ PyArray_ConcatenateArrays(int narrays, PyArrayObject **arrays, int axis,
             return NULL;
         }
 
+        initial_byteorder = PyArray_DESCR(arrays[0])->byteorder;
+        for (iarrays = 0; iarrays < narrays; ++iarrays) {
+            iterative_byteorder = PyArray_DESCR(arrays[iarrays])->byteorder;
+            if (iterative_byteorder != initial_byteorder) {
+                single_byteorder = 0;
+            }
+        }
```

Comment:
we already have loops that do this iteration, and I did have success fusing this into those loops if that is preferred; obviously it doesn't change the asymptotic complexity

Reply:
(we could also `break` early I suppose, if that condition is ever satisfied)